server/src/main/java/org/elasticsearch/action/search/QueryPhaseResultConsumer.java

```diff
@@ -140,7 +140,7 @@ public void consumeResult(SearchPhaseResult result, Runnable next) {
         }
         SearchPhaseController.ReducedQueryPhase reducePhase = controller.reducedQueryPhase(results.asList(), aggsList,
             topDocsList, topDocsStats, pendingMerges.numReducePhases, false, aggReduceContextBuilder, performFinalReduce);
-        if (hasAggs) {
+        if (hasAggs && reducePhase.aggregations != null) {
```

Comment:
Is it needed both conditions? I would expect that if reducePhase.aggregations is not null, then hasAggs is always true?

Reply:
I added a comment in https://github.com/elastic/elasticsearch/pull/66280/commits/baabbbd38f395b7ca6adacd590a5475db647f168, it can be null if all shards failed.